01-atoti-academy/introduction-to-atoti/main.ipynb

```diff
@@ -206,7 +248,6 @@
     "    user_content_storage=\"./content\",\n",
     "    port=9092,\n",
     "    java_options=[\"-Xms1G\", \"-Xmx10G\"],\n",
-    "    app_extensions=tt.ADVANCED_APP_EXTENSION,\n",
```

Comment:
I had to take out this line as it was causing the following issue:

```
Closing existing "Unnamed" session to create the new one.
---------------------------------------------------------------------------
Py4JJavaError                             Traceback (most recent call last)
File ~/anaconda3/lib/python3.10/site-packages/atoti/_local_session.py:231, in LocalSession.__init__(self, address, config, distributed, enable_py4j_auth, license_key, name, plugins, py4j_server_port, start_application, wrap_start_error)
    230 try:
--> 231     self._start_application()
    232 except Py4JError as error:

File ~/anaconda3/lib/python3.10/site-packages/atoti/_local_session.py:297, in LocalSession._start_application(self)
    296 def _start_application(self) -> None:
--> 297     self._java_api.start_application(self._config)

File ~/anaconda3/lib/python3.10/site-packages/atoti/_java_api.py:309, in JavaApi.start_application(self, config)
    308 json_config = serialize_config_to_json(config)
--> 309 self.java_session.startServer(json_config)

File ~/anaconda3/lib/python3.10/site-packages/py4j/java_gateway.py:1322, in JavaMember.__call__(self, *args)
   1321 answer = self.gateway_client.send_command(command)
-> 1322 return_value = get_return_value(
   1323     answer, self.gateway_client, self.target_id, self.name)
   1325 for temp_arg in temp_args:

File ~/anaconda3/lib/python3.10/site-packages/py4j/protocol.py:326, in get_return_value(answer, gateway_client, target_id, name)
    325 if answer[1] == REFERENCE_TYPE:
--> 326     raise Py4JJavaError(
    327         "An error occurred while calling {0}{1}{2}.\n".
    328         format(target_id, ".", name), value)
    329 else:

Py4JJavaError: An error occurred while calling t.startServer.
: com.activeviam.fwk.ActiveViamRuntimeException: None of the active plugins support the "app_extensions" config key.
	at io.atoti.configuration.SessionConfigDeserializer.lambda$checkKeysSupportedByActivePlugins$1(SessionConfigDeserializer.java:44)
	at java.base/java.util.LinkedHashMap$LinkedKeySet.forEach(Unknown Source)
	at io.atoti.configuration.SessionConfigDeserializer.checkKeysSupportedByActivePlugins(SessionConfigDeserializer.java:42)
	at io.atoti.configuration.SessionConfigDeserializer.deserializeConfig(SessionConfigDeserializer.java:24)
	at io.atoti.SessionImpl.configure(SessionImpl.java:108)
	at io.atoti.ServerManager.processConfiguration(ServerManager.java:96)
	at io.atoti.ServerManager.startServer(ServerManager.java:56)
	at io.atoti.SessionImpl.startServer(SessionImpl.java:197)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
	at py4j.Gateway.invoke(Gateway.java:282)
	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
	at py4j.commands.CallCommand.execute(CallCommand.java:79)
	at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
	at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
	at java.base/java.lang.Thread.run(Unknown Source)


The above exception was the direct cause of the following exception:

RuntimeError                              Traceback (most recent call last)
Cell In[4], line 1
----> 1 session = tt.Session(
      2     user_content_storage="./content",
      3     port=9092,
      4     java_options=["-Xms1G", "-Xmx10G"],
      5     app_extensions=tt.ADVANCED_APP_EXTENSION,
      6 )

File ~/anaconda3/lib/python3.10/site-packages/atoti/_typecheck.py:18, in typecheck.<locals>.wrapper(*args, **kwargs)
     16 memo = _CallMemo(func=function, args=args, kwargs=kwargs)
     17 check_argument_types(memo)
---> 18 return function(*args, **kwargs)

File ~/anaconda3/lib/python3.10/site-packages/atoti/_telemetry/track_calls.py:68, in create_decorator.<locals>.decorator.<locals>.wrapper(*args, **kwargs)
     65 call_time = perf_counter()
     67 try:
---> 68     return function(*args, **kwargs)
     69 except Exception as error:
     70     with suppress(  # Do nothing to let the previous error be the one presented to the user.
     71         Exception
     72     ):

File ~/anaconda3/lib/python3.10/site-packages/atoti/session.py:259, in Session.__init__(self, name, app_extensions, authentication, branding, client_certificate, extra_jars, https, i18n, java_options, jwt, logging, port, ready, same_site, user_content_storage, **kwargs)
    235 if private_parameters.start_application:
    236     config = private_parameters.config or SessionConfig(
    237         app_extensions=app_extensions,
    238         authentication=create_authentication_config(authentication)
   (...)
    256         else user_content_storage,
    257     )
--> 259     super().__init__(
    260         address=None,
    261         config=config,
    262         distributed=False,
    263         enable_py4j_auth=private_parameters.enable_py4j_auth,
    264         license_key=private_parameters.license_key,
    265         name=name,
    266         plugins=private_parameters.plugins,
    267         py4j_server_port=private_parameters.py4j_server_port,
    268         start_application=private_parameters.start_application,
    269         wrap_start_error=private_parameters.wrap_start_error,
    270     )
    271 else:
    272     super().__init__(
    273         name=name,
    274         address=private_parameters.address,
   (...)
    282         wrap_start_error=private_parameters.wrap_start_error,
    283     )

File ~/anaconda3/lib/python3.10/site-packages/atoti/_local_session.py:233, in LocalSession.__init__(self, address, config, distributed, enable_py4j_auth, license_key, name, plugins, py4j_server_port, start_application, wrap_start_error)
    231         self._start_application()
    232     except Py4JError as error:
--> 233         raise RuntimeError(
    234             f"An error occurred while starting the session. The logs are available at {self.logs_path}"
    235         ) from error
    236 else:
    237     self._start_application()

RuntimeError: An error occurred while starting the session. The logs are available at /Users/aya/.atoti/1701893987_74NWR3/logs/server.log
```

And the associated `/Users/aya/.atoti/1701893987_74NWR3/logs/server.log` states the following:

```
(base) aya@aya logs % cat startup.log 
Dec 06, 2023 3:19:48 PM com.qfs.platform.share.SharePlatform <init>
WARNING: The platform is not supported. Everything will behave as on an UMA system.
Dec 06, 2023 3:19:53 PM ActiveViam License log
INFO: Activating ActiveViam Licensing Agent.
Dec 06, 2023 3:19:53 PM com.activeviam.activation.impl.LicenseManager a
INFO: License has been found from system property -Dactivepivot.license=/Users/aya/anaconda3/lib/python3.10/site-packages/atoti/data/community.lic
Dec 06, 2023 3:19:53 PM ActiveViam License log
INFO: 
*************** Platform Information *****************
System Time: December 6, 2023
Number of Processors: 14
MAC address: 60-3E-5F-67-8C-38
IP address: 192.168.90.74
Hostname: aya.local
******************************************************

********** ActivePivot License Information ***********
License ID: 88baf88d-4f22-4b23-bc4c-5ec99fe85ee6
Client name: atoti community
End Date: Tue Feb 13 00:00:00 EST 2024
Maximum Number of Processors: 0
Restricted MAC address: -
Restricted IP address: -
Restricted Hostname: -
On Demand: true
On Demand License Server: AWS_EU_WEST_1_PROD
Target Environment: -
Datastore Enabled: true
DirectQuery Enabled: false
ActiveUI Enabled: true
ActiveMonitor Enabled: true
Collateral Optimizer Enabled: false
******************************************************

ActivePivot Version: 6.0.10-20231030-111721
Dec 06, 2023 3:19:53 PM ActiveViam License log
INFO: Connection to license server succeeded.
Dec 06, 2023 3:19:53 PM ActiveViam License log
INFO: License loaded and tested successfully.
Dec 06, 2023 3:19:53 PM com.quartetfs.fwk.contributions.impl.AnnotationContributionProvider parseAnnotations
WARNING: The following class did not bring any contribution to the Registry: class com.qfs.messenger.impl.DistributedAgentExtendedPlugin
Dec 06, 2023 3:19:53 PM com.quartetfs.fwk.contributions.impl.AnnotationContributionProvider parseAnnotations
WARNING: The following class did not bring any contribution to the Registry: class com.quartetfs.biz.pivot.context.drillthrough.impl.CaptionDrillthroughColumn
Dec 06, 2023 3:19:53 PM com.quartetfs.fwk.contributions.impl.AnnotationContributionProvider parseAnnotations
WARNING: The following class did not bring any contribution to the Registry: class com.qfs.distribution.security.impl.DistributedSecurityManagerExtendedPlugin
Dec 06, 2023 3:19:53 PM com.quartetfs.fwk.contributions.impl.AnnotationContributionProvider parseAnnotations
WARNING: The following class did not bring any contribution to the Registry: class com.quartetfs.fwk.ordering.impl.ComparatorExtendedPlugin
Dec 06, 2023 3:19:53 PM com.quartetfs.fwk.contributions.impl.AnnotationContributionProvider parseAnnotations
WARNING: The following class did not bring any contribution to the Registry: class com.activeviam.pivot.rest.dataexport.converters.selectconverter.impl.MdxSelectToArrowConverter
Dec 06, 2023 3:19:53 PM com.quartetfs.fwk.contributions.impl.AnnotationContributionProvider parseAnnotations
WARNING: The following class did not bring any contribution to the Registry: class com.quartetfs.biz.pivot.context.drillthrough.impl.NameDrillthroughColumn
Dec 06, 2023 3:19:53 PM com.quartetfs.fwk.contributions.impl.AnnotationContributionProvider parseAnnotations
WARNING: The following class did not bring any contribution to the Registry: class com.activeviam.pivot.json.dataexport.jsonoutputconfiguration.impl.JsonArrowOutputConfiguration
Dec 06, 2023 3:19:53 PM com.quartetfs.fwk.contributions.impl.AnnotationContributionProvider parseAnnotations
WARNING: The following class did not bring any contribution to the Registry: class com.qfs.distribution.impl.MultiVersionDistributedActivePivotExtendedPlugin
Dec 06, 2023 3:19:53 PM com.quartetfs.fwk.contributions.impl.AnnotationContributionProvider parseAnnotations
WARNING: The following class did not bring any contribution to the Registry: class com.quartetfs.biz.pivot.context.drillthrough.impl.MemberValueDrillthroughColumn
Dec 06, 2023 3:19:54 PM com.quartetfs.fwk.impl.QFSRegistry getPlugin
WARNING: After initialization Plugin (for interface com.activeviam.activepivot.core.intf.internal.structure.IPivotStructureUpdaterFromMeasure) does not contain any plugin value.
Dec 06, 2023 3:19:54 PM com.quartetfs.fwk.impl.QFSRegistry getPlugin
WARNING: After initialization Plugin (for interface com.qfs.store.service.IUpdateWhereProcedureFactory) does not contain any plugin value.
Dec 06, 2023 3:19:54 PM io.atoti.util.impl.ServerUtils startPy4JServer
INFO: Starting Py4J Gateway
Dec 06, 2023 3:19:54 PM io.atoti.util.impl.ServerUtils$GatewayListener serverStarted
INFO: Py4J server started on port 25333 with auth token umA1fI6uTPg8U8Ct6QhoFBpjIleqzAZnNMQhIxAnuYI=
```

Reply:
Also, note that the error output suggests that we need to see the `server.log`, but it actually seems to be named `startup.log`. Very minor detail but marking for consistency.

```
(base) aya@aya logs % pwd
/Users/aya/.atoti/1701893987_74NWR3/logs
(base) aya@aya logs % ls -l
total 16
-rw-r--r--  1 aya  staff  4694 Dec  6 15:19 startup.log
```